code/modules/clothing/glasses/hud.dm

```diff
@@ -45,17 +49,75 @@
 
 /obj/item/clothing/glasses/hud/emp_act(severity)
 	. = ..()
-	if(obj_flags & EMAGGED || . & EMP_PROTECT_SELF)
+	if(obj_flags & OBJ_EMPED || . & EMP_PROTECT_SELF)
 		return
-	obj_flags |= EMAGGED
+	obj_flags |= OBJ_EMPED
 	desc = "[desc] The display is flickering slightly."
+	addtimer(CALLBACK(src, .proc/reset_emp), rand(1200 / severity, 600 / severity))
+	//If we aren't glitching out already, start glitching
+	if(!(obj_flags & EMAGGED))
+		start_glitch()
+
+/obj/item/clothing/glasses/hud/proc/reset_emp()
+	obj_flags &= ~OBJ_EMPED
+	//If we aren't emagged, stop glitching
+	if(obj_flags & EMAGGED)
+		return
+	desc = initial(desc)
+	stop_glitch()
 
 /obj/item/clothing/glasses/hud/emag_act(mob/user)
 	if(obj_flags & EMAGGED)
 		return
 	obj_flags |= EMAGGED
 	to_chat(user, "<span class='warning'>PZZTTPFFFT</span>")
 	desc = "[desc] The display is flickering slightly."
+	//If we aren't already glitching out, start glitching
+	if(!(obj_flags & OBJ_EMPED))
+		start_glitch()
+
+/obj/item/clothing/glasses/hud/proc/start_glitch()
+	if(ismob(loc))
+		var/mob/M = loc
+		//Remove old glitching hud
+		if(glitching_hud)
+			glitching_hud.transform = matrix()
+			glitching_hud.filters = null
+			glitching_hud.color = null
+			glitching_hud = null
+		//Get a new glitching hud
+		glitching_hud = locate(/atom/movable/screen/plane_master/data_hud) in M.client?.screen
+
+	START_PROCESSING(SSobj, src)
+
+/obj/item/clothing/glasses/hud/proc/stop_glitch()
+	if(glitching_hud)
+		glitching_hud.transform = matrix()
+		glitching_hud.filters = null
+		glitching_hud.color = null
+		glitching_hud = null
+
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/clothing/glasses/hud/process(delta_time)
+	if(!glitching_hud)
+		return
```

Comment:
If we don't have hud there something has gone terribly wrong and you should probably take that into account.

Reply:
A hud being glitched and broken is completely outside the bounds of what this is doing and shouldn't be fixed in this proc.

Reply:
Oh I get what you mean now, this should never happen but it is still there just in case otherwise we will be throwing a lot of runtimes, and if it does happen it will solve itself when dropped and picked up again.